cumulusci/robotframework/pageobjects/BasePageObjects.py

```diff
@@ -0,0 +1,270 @@
+            locator = "//ul[@class='errorsList']//li[contains(., \"{}\")]".format(
+                message
+            )
+            self.selenium.page_should_contain_element(
+                locator,
+                'The page did not contain an error with the text "{}"'.format(message),
+            )
+
+    @capture_screenshot_on_error
+    def populate_field(self, name, value):
+        """Populate a field on the dialog form
+
+        Name must the the label of a field as it appears on the dialog form.
+
+        Example
+
+        | Populate field  First Name  Connor
+        | Populate field  Last Name   MacLeod
+        """
+        # For now, call the same keyword in Salesforce.py. Eventually
+        # that keyword may get moved here and deprecated from that
+        # library.
+        self.salesforce.populate_field(name, value)
+
+    @capture_screenshot_on_error
+    def populate_form(self, *args, **kwargs):
+        """Populate the dialog form
+
+        Arguments are of the form key=value, where 'key' represents
+        a field name as it appears on the form (specifically, the text
+        of a label with the class 'uiLabel').
+
+        Example:
+
+        | Populate form
+        | ...  First Name=Connor
+        | ...  Last Name=MacLeod
+        """
+        # For now, call the same keyword in Salesforce.py. Eventually
+        # that keyword may get moved here and deprecated from that
+        # library.
+        self.salesforce.populate_form(*args, **kwargs)
+
+    @capture_screenshot_on_error
+    def wait_until_dialog_is_closed(self, timeout=None):
+        """Waits until the dialog is no longer visible
+
+        If the dialog isn't open, this will not throw an error.
+        """
+        locator = "//div[contains(@class, 'uiModal')]"
+
+        self.selenium.wait_until_page_does_not_contain_element(locator)
+
+
+@pageobject("Home")
+class HomePage(BasePage):
+    """A page object representing the home page of an object.
+
+    When going to the Home page, you need to specify the object name.
+
+    Note: The home page of an object may automatically redirect you to
+    some other page, such as a Listing page. If you are working with
+    such a page, you might need to use `page should be` to load the
+    keywords for the page you expect to be redirected to.
+
+    Example
+
+    | Go to page  Home  Contact
+
+    """
+
+    def _go_to_page(self):
+        url_template = "{root}/lightning/o/{object_name}/home"
+        url = url_template.format(
+            root=self.cumulusci.org.lightning_base_url, object_name=self.object_name
+        )
+        self.selenium.go_to(url)
+        self.salesforce.wait_until_loading_is_complete()
+
+    def _is_current_page(self):
+        self.selenium.location_should_contain(
+            "/lightning/o/{}/home".format(self.object_name)
+        )
+
+
+@pageobject("Detail")
+class DetailPage(BasePage):
+    """A page object representing the standard Detail page.
```

Comment:
I think this is typically called a Record Page in Salesforce documentation.

Reply:
It's had this name since we first introduced page objects. Changing it now would break a lot of tests.

Reply:
Can we rename it to RecordPage and use that in docs, but also keep it available as DetailPage as a backward-compatible alias?

Reply:
Yes, we could. I think I'll save that for a separate work item, since this specific change was just a result of moving code from one file to another.